lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2628,52 +2633,104 @@ void processAnnotationString(IntrinsicInst *II, std::string &AnnotationString) {
       processOptionalAnnotationInfo(C, AnnotationString);
 }
 
+// Try to parse the annotation decoration values in a string. These values must
+// be separated by a "," and must be either a word (including numbers) or a
+// quotation mark enclosed string.
+static bool tryParseAnnotationDecoValues(StringRef ValueStr,
+                                         std::vector<std::string> &ParsedArgs) {
+  unsigned ValueStart = 0;
+  bool ParsingStrLit = false;
+  for (unsigned I = 0; I < ValueStr.size(); ++I) {
+    const char CurrentC = ValueStr[I];
+    if (ParsingStrLit) {
+      if (CurrentC == '"') {
+        ParsingStrLit = false;
+        ParsedArgs.push_back(ValueStr.substr(ValueStart, I - ValueStart).str());
+        if (I + 1 != ValueStr.size() && ValueStr[I + 1] != ',')
```

Comment:
I'd personally like to have some comments on exit conditions in this function - nice to mention why (maybe with examples?) the parsed string is not valid.

Reply:
I've added some comments documenting most of the parsing. Is this along the lines of what you were thinking?

Reply:
Absolutely, thanks! It'll simplify the code support in future.